pandas/_libs/tslibs/offsets.pyx

```diff
@@ -1140,7 +1141,9 @@ cdef class RelativeDeltaOffset(BaseOffset):
 
             weeks = kwds.get("weeks", 0) * self.n
             if weeks:
-                dt64other = dt64other + Timedelta(days=7 * weeks)
+                delta = Timedelta(days=7 * weeks)
+                td = (<_Timedelta>delta)._as_reso(reso)
```

Comment:
In the future this cast isn't needed once `_as_reso` is public?

Reply:
needed as long as it is cdef instead of cpdef